frame/ibc-transfer/src/lib.rs

```diff
@@ -0,0 +1,576 @@
+		channel_id_from_bytes, connection_id_from_bytes, port_id_from_bytes, IbcTrait,
+		OpenChannelParams,
+	};
+	use primitives::currency::CurrencyId;
+	use sp_runtime::{traits::IdentifyAccount, AccountId32};
+
+	#[derive(
+		frame_support::RuntimeDebug,
+		PartialEq,
+		Eq,
+		scale_info::TypeInfo,
+		Encode,
+		Decode,
+		Copy,
+		Clone,
+		Default,
+		codec::MaxEncodedLen,
+	)]
+	pub struct PalletParams {
+		pub send_enabled: bool,
+		pub receive_enabled: bool,
+	}
+
+	#[derive(
+		frame_support::RuntimeDebug,
+		PartialEq,
+		Eq,
+		scale_info::TypeInfo,
+		Encode,
+		Decode,
+		Clone,
+		Default,
+	)]
+	pub struct TransferParams {
+		/// Valid utf8 string bytes
+		pub to: Vec<u8>,
+		/// Source channel on host chain
+		pub source_channel: Vec<u8>,
+		/// Timestamp at which this packet should timeout in counterparty in nanoseconds
+		pub timeout_timestamp: u64,
+		/// Block height at which this packet should timeout on counterparty
+		pub timeout_height: u64,
+		/// Revision number, only needed when making a transfer to a parachain
+		/// in which case this should be the para id
+		pub revision_number: Option<u64>,
+	}
+
+	/// Configure the pallet by specifying the parameters and types on which it depends.
+	#[pallet::config]
+	pub trait Config: frame_system::Config + balances::Config + DeFiComposableConfig {
+		/// The overarching event type.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		/// A type for creating local asset Ids
+		type CurrencyFactory: CurrencyFactory<
+			<Self as DeFiComposableConfig>::MayBeAssetId,
+			<Self as DeFiComposableConfig>::Balance,
+		>;
+
+		/// Account Id Conversion from SS58 string or hex string
+		type AccountIdConversion: TryFrom<Signer>
+			+ IdentifyAccount<AccountId = Self::AccountId>
+			+ Clone;
+		/// A type that allows us map ibc assets to local assets
+		type AssetRegistry: RemoteAssetRegistryMutate + RemoteAssetRegistryInspect;
+		/// MultiCurrency System
+		type MultiCurrency: Transfer<
+				Self::AccountId,
+				Balance = <Self as DeFiComposableConfig>::Balance,
+				AssetId = <Self as DeFiComposableConfig>::MayBeAssetId,
+			> + Mutate<
+				Self::AccountId,
+				Balance = <Self as DeFiComposableConfig>::Balance,
+				AssetId = <Self as DeFiComposableConfig>::MayBeAssetId,
+			> + Inspect<
+				Self::AccountId,
+				Balance = <Self as DeFiComposableConfig>::Balance,
+				AssetId = <Self as DeFiComposableConfig>::MayBeAssetId,
+			>;
+		type IbcHandler: ibc_trait::IbcTrait;
+		#[pallet::constant]
+		type PalletId: Get<PalletId>;
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		TokenTransferInitiated {
+			from: <T as frame_system::Config>::AccountId,
+			to: Vec<u8>,
+			amount: <T as DeFiComposableConfig>::Balance,
```

Comment:
Not using the `MultiCurrency` trait directly here, using these instead `frame_support::traits::fungibles::{Transfer, Mutate}` to handle minting and token transfers, because these are the traits implemented by `pallet-assets` to handle the said operations.
On the other hand the `DefiComposableConfig` trait is meant to be used  to specify the `Balance` and `AssetId` types used in the runtime, that's why I'm delegating to it in this pallet.